servers/apps/test/malloc_test.c

```diff
@@ -0,0 +1,30 @@
+#include <resea/malloc.h>
+#include <resea/printf.h>
+#include <string.h>
+#include "test.h"
+#define NUM_PTRS 16
+
+void malloc_test(void) {
+    malloc_init();
+    void *ptr[NUM_PTRS];
+    memset(ptr, 0, sizeof(ptr));
+    int sz = 1;
+
+    for (size_t i = 0; i < NUM_PTRS - 1; i++) {
+        ptr[i] = malloc(sz << i);
+        TEST_ASSERT(ptr[i] != NULL);
+    }
+
+    ptr[NUM_PTRS - 1] = malloc((1 << 15) + 8);
+    for (size_t i = 0; i < NUM_PTRS; i++) {
+        free(ptr[i]);
```

Comment:
Before free'ing, do `memset(ptr[i], 0xaa, size);`. A heap overflow detection in `free()` could catch bugs in malloc.

Reply:
Will the size parameter here be equal to the "size" of the chunk? A chunk has some metadata also right, and that is before the actual data.

Reply:
Yes `size` for `memset` should be equal to the size specified in `malloc(size)` above.

> A chunk has some metadata also right? And that is before the actual data.

You're correct. Adding memset ensures that malloc does not allocate too short chunk for the requested size.

Reply:
Okay, thanks for clarifying. I have made the change.